### PR TITLE
feat: generate main and sub boards

### PR DESCRIPTION
**Pull request by nunocaseiro**
Relates to #137

## Screenshots 

image

## Proposed Changes

  - Generate main board
  - Generate sub boards from the xgeeks team
  - Quick edit settings (changing sub teams names or max number of people in each team)
  - Create board request to the backend
  - Feature to replace the responsible of each team

### CODE REVIEW

File: frontend/hooks/useCreateBoard.tsx [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+      }
+
+      for (let i = 0; i < maxTeams; i++) {
+        const newBoard = generateSubBoard(i + 1);
+        splitedUsers[i][Math.floor(Math.random() * splitedUsers[i].length)].role =
+          BoardUserRoles.RESPONSIBLE;
+        newBoard.users = splitedUsers[i];
+        subBoards.push(newBoard);
+      }
+      return subBoards;
```
Comment (RubenMCCarreira) [inline frontend/hooks/useCreateBoard.tsx:91]:
```suggestion
const subBoards: BoardToAdd[] = [];
const splitedUsers: BoardUserToAdd[][] = [];

let availableUsers;

new Array (teamMembers.length).forEach((_, i) => {
  const teamUser = getRandomUser(availableUsers);
  if (splitedUsers[i]?.length === maxUsers) break;

  if (i >= maxTeams) i = 0;
  splitedUsers.push({
    user: teamUser.user,
    role: BoardUserRoles.MEMBER,
    votesCount: 0,
  })

  availableUsers = teamMembers.filter((user) => user.user._id !== teamUser.user._id);
})

new Array(maxTeams).forEach((_, i) => {
  const newBoard = generateSubBoard(i + 1);
  splitedUsers[i][Math.floor(Math.random() * splitedUsers[i].length)].role =
    BoardUserRoles.RESPONSIBLE;
  newBoard.users = splitedUsers[i];
  subBoards.push(newBoard);
})

return subBoards;
```

Comment (RubenMCCarreira) [reply frontend/hooks/useCreateBoard.tsx:91]: try this code pls, and check if it does the expected

Comment (nunocaseiro) [reply frontend/hooks/useCreateBoard.tsx:91]: Thanks for the tip but the break can't be used in a for each

Comment (RubenMCCarreira) [reply frontend/hooks/useCreateBoard.tsx:91]: replace for a `return;`

Comment (nunocaseiro) [reply frontend/hooks/useCreateBoard.tsx:91]: there is no way to break a for each

Comment (nunocaseiro) [reply frontend/hooks/useCreateBoard.tsx]: The first loop was wrong implemented. I noticed the `let j` was missing. However I tried with your suggestion `new Array` but I can't change de index in a for each so I cannot achieve the same behaviour. Is this regular for loop a wrong approach? I can try to find a better way to do this

Comment (nunocaseiro) [reply frontend/hooks/useCreateBoard.tsx]: I simplified the for loop and i also used your suggestion for the last loop to generate sub boards

**Review by RubenMCCarreira — APPROVED** [review]

**Pull request merged**